File: app/src/main/java/dev/ragnarok/fenrir/settings/NightMode.java
package dev.ragnarok.fenrir.settings;

import androidx.annotation.IntDef;
import androidx.appcompat.app.AppCompatDelegate;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@IntDef({NightMode.FOLLOW_SYSTEM, NightMode.DISABLE, NightMode.ENABLE, NightMode.AUTO})
@Retention(RetentionPolicy.SOURCE)
public @interface NightMode {
    int DISABLE = AppCompatDelegate.MODE_NIGHT_NO;
    int ENABLE = AppCompatDelegate.MODE_NIGHT_YES;
    int AUTO = AppCompatDelegate.MODE_NIGHT_AUTO_BATTERY;
    int FOLLOW_SYSTEM = AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM;
}

File: app_filegallery/src/main/kotlin/dev/ragnarok/filegallery/settings/NightMode.kt
package dev.ragnarok.filegallery.settings

import androidx.annotation.IntDef
import androidx.appcompat.app.AppCompatDelegate

@IntDef(NightMode.FOLLOW_SYSTEM, NightMode.DISABLE, NightMode.ENABLE, NightMode.AUTO)
@Retention(
    AnnotationRetention.SOURCE
)
annotation class NightMode {
    companion object {
        const val DISABLE = AppCompatDelegate.MODE_NIGHT_NO
        const val ENABLE = AppCompatDelegate.MODE_NIGHT_YES
        const val AUTO = AppCompatDelegate.MODE_NIGHT_AUTO_BATTERY
        const val FOLLOW_SYSTEM = AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
    }
}